openmetadata-service/src/main/java/org/openmetadata/service/resources/dqtests/TestCaseResource.java

```diff
@@ -1000,6 +1000,31 @@ public TableData getFailedRowsData(
     return repository.getSampleData(testCase, authorizePII);
   }
 
+  @DELETE
+  @Path("/{id}/failedRowsSample")
+  @Operation(
+      operationId = "deleteFailedRowsSample",
+      summary = "Delete failed rows sample data",
+      description = "Delete a sample of failed rows for this test case.",
+      responses = {
+        @ApiResponse(responseCode = "200", description = "OK"),
+        @ApiResponse(
+            responseCode = "404",
+            description = "Failed rows sample data for test case {id} is not found.")
+      })
+  public Response deleteFailedRowsData(
+      @Context UriInfo uriInfo,
+      @Context SecurityContext securityContext,
+      @Parameter(description = "Id of the table", schema = @Schema(type = "UUID")) @PathParam("id")
+          UUID id) {
+    OperationContext operationContext =
+        new OperationContext(entityType, MetadataOperation.VIEW_SAMPLE_DATA);
```

Comment:
@sushi30 we should check DELETE_SAMPLE_DATA operation here. or Owner of the test or Owner of the test that this table belongs to

Reply:
Also we should introduce a separate permission here.
1. Who should be able to view sample data for a failed test 
  2.  This should only be for Admins, Owner of table, owner of test
3. How we grant sample data permission
   4.  We shouldn't mix view sample data of table to sample failed rows of test case. Both serve different purpose

Reply:
> Also we should introduce a separate permission here.

@harshach IMO the principle here is the same in both cases - though use case are different. I can't think of a use case where one would be allowed to view sample data for a table but not the sample data of the failed rows for the same table. 

Do you have any use case in mind, I might be missing something?

Reply:
@harshach while they serve different purposes, the permissions are essentially for the same data (i.e: viewing data from the underlying table). the connection between the data and the test is only from OM perspective.

in terms of governance policy, if a user has certain permissions to view the data in the context of a profiler result, is there a reason they should have different access in the context of a test case result?